src/opm/parser/eclipse/EclipseState/Schedule/KeywordHandlers.cpp

```diff
@@ -1855,6 +1855,26 @@ namespace {
         }
     }
 
+    void Schedule::handleWPAVEDEP(const HandlerContext& handlerContext, const ParseContext& parseContext, ErrorGuard& errors) {
+        for (const auto& record : handlerContext.keyword) {
+            const std::string& wellNamePattern = record.getItem<ParserKeywords::WPAVEDEP::WELL>().getTrimmedString(0);
+            const auto well_names = wellNames(wellNamePattern, handlerContext.currentStep);
+
+            if (well_names.empty())
+                invalidNamePattern(wellNamePattern, handlerContext.currentStep, parseContext, errors, handlerContext.keyword);
+
+            const auto& item = record.getItem<ParserKeywords::WPAVEDEP::REFDEPTH>();
+            if (item.hasValue(0)) {
+                auto ref_depth = item.getSIDouble(0);
+                for (const auto& well_name : well_names) {
+                    auto well = std::shared_ptr<Well>(new Well( this->getWell(well_name, handlerContext.currentStep)));
```

Comment:
Just a quick heads-up.  This (`shared_ptr<Well>(new Well(...))`) is better written as
```C++
make_shared<Well>(...)
```
Both because it avoids a bare `new` call and because `make_shared` enables nifty optimisation that stores the shared_ptr's control block in the same allocation as the object.

Reply:
I'll update